pandas/tests/frame/methods/test_astype.py

```diff
@@ -787,3 +787,18 @@ def test_frame_astype_no_copy():
 
     assert result.a.dtype == pd.Int16Dtype()
     assert np.shares_memory(df.b.values, result.b.values)
+
+
+def test_datetime64_eq_string():
+    # Issue https://github.com/pandas-dev/pandas/issues/45506
+    # Catch OverflowError when comparing datetime64 and string
+    data = [
+        {"a": "2015-07-01", "b": "08335394550"},
+        {"a": "2015-07-02", "b": "+49 (0) 0345 300033"},
+        {"a": "2015-07-03", "b": "+49(0)2598 04457"},
+        {"a": "2015-07-04", "b": "0741470003"},
+        {"a": "2015-07-05", "b": "04181 83668"},
+    ]
+    dtypes = {"a": "datetime64[ns]", "b": "string"}
+    df = DataFrame(data=data).astype(dtypes)
+    tm.assert_series_equal(df["a"].eq(df["b"]), Series([False] * 5))
```

Comment:
can you do
```
result = 
expected = 
tm.assert_series_equal.....
```

i see this is actually a comparison test, ok then let's move it to simliar in series
maybe in pandas/tests/series/methods/test_compare.py or in indexing. have to look around a bit.